pkg/api/v1/httptools.go

```diff
@@ -25,6 +25,23 @@ func (r *Router) parseTenantID(c echo.Context) (string, error) {
 	return tenantID, nil
 }
 
+// parseUUID parses and validates a UUID from the request path if the path param is found
+// returns UUID or empty string in the case where path param is not found
+func (r *Router) parseUUID(c echo.Context, path string) (string, error) {
+	var id string
+	if err := echo.PathParamsBinder(c).String(path, &id).BindError(); err != nil {
+		return "", err
+	}
+
+	if id != "" {
+		if _, err := uuid.Parse(id); err != nil {
+			return "", ErrInvalidUUID
+		}
+	}
+
+	return id, nil
+}
```

Comment:
Did you give some thought to this function returning an error if the path param wasn't found?  You could `errors.Is()` in the calling function in that case.

Reply:
Yeah, that works. Updated 😄